packages/java-parser/src/tokens.js

```diff
@@ -32,17 +33,46 @@ FRAGMENT("HexDigits", "{{HexDigit}}(({{HexDigit}}|'_')*{{HexDigit}})?");
 FRAGMENT("FloatTypeSuffix", "[fFdD]");
 FRAGMENT("LineTerminator", "(\\x0A|(\\x0D(\\x0A)?))");
 
+function matchJavaIdentifier(char, startOffset) {
+  let endOffset = startOffset;
+  let charCode = char.codePointAt(endOffset);
+
+  // We verifiy if the first character is from one of these categories
+  // Corresponds to the isJavaIdentifierStart function from Java
+  if (chars.firstIdentChar.has(charCode)) {
+    endOffset++;
+    charCode = char.charCodeAt(endOffset);
+  }
+
+  // We verify if the remaining characters is from one of these categories
+  // Corresponds to the isJavaIdentifierPart function from Java
+  while (chars.restIdentChar.has(charCode)) {
+    endOffset++;
+    charCode = char.codePointAt(endOffset);
+  }
+
+  // No match, must return null to conform with the RegExp.prototype.exec signature
+  if (endOffset === startOffset) {
+    return null;
+  }
+  const matchedString = char.substring(startOffset, endOffset);
+  // according to the RegExp.prototype.exec API the first item in the returned array must be the whole matched string.
+  return [matchedString];
+}
+
 const Identifier = createTokenOrg({
   name: "Identifier",
-  // TODO: Align with the spec, Consider generating the regExp for Identifier
-  //       as done in Esprima / Acorn
-  pattern: /[a-zA-Z_\\$][a-zA-Z_\\$0-9]*/
+  pattern: { exec: matchJavaIdentifier },
+  line_breaks: false,
+  start_chars_hint: Array.from(chars.firstIdentChar, x =>
+    String.fromCodePoint(x)
```

Comment:
the start_chars_hints is utf16
see: https://sap.github.io/chevrotain/documentation/4_3_1/interfaces/itokenconfig.html#start_chars_hint

So I think you need to use String.fromCharCode